docs/getting_started.rst

```diff
@@ -31,105 +39,467 @@ example to construct such a store for the local filesystem.
-There is also ``iter`` that supports reading the dataset on
-a per-partition base. For larger dataset, ``dask`` can be used
-to work on datasets in parallel or even in a cluster by using
-``distributed`` as the backend for ``dask``.
+Writing dataset to storage
+===========================
+
+Now that we have our data and the storage location, we can persist the dataset.
+To do so, we will use :func:`kartothek.io.eager.store_dataframes_as_dataset`
+to store a ``DataFrame`` we already have in memory.
 
 .. ipython:: python
-   :okwarning:
 
-   from kartothek.io.eager import store_dataframes_as_dataset
-   dm = store_dataframes_as_dataset(
-      store, "a_unique_dataset_identifier", df, metadata_version=4
-   )
-   dm
+    from kartothek.io.eager import store_dataframes_as_dataset
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "a_unique_dataset_identifier",
+        df,
+    )
+    dm
+
+.. admonition:: Storage backends
+
+    The import path of this function already gives us a hint about the general
+    structuring of the ``kartothek`` modules. In :mod:`kartothek.io` we have all
+    the building blocks to build data pipelines that read and write from/to storages.
+    The next module level (e.g. ``eager``) describes the scheduling backend.
+
+    The scheduling backends supported by kartothek are:
+
+    - ``eager`` runs all execution immediately and on the local machine.
+    - ``iter`` executes operations on the dataset on a per-partition basis.
+      The standard format to read/store dataframes in ``iter`` is by providing
+      a generator of dataframes.
+    - ``dask`` is suitable for larger datasets. It can be used to work on datasets in
+      parallel or even in a cluster by using ``dask.distributed`` as the backend.
+      There are also ``dask.bag`` and ``dask.dataframe`` which support I/O operations
+      for the respective `dask`_ objects.
 
 After calling :func:`~kartothek.io.eager.store_dataframes_as_dataset`,
-a :class:`kartothek.core.dataset.DatasetMetadata` object is returned. This is the main
-class holding all information about the parts and schema of the dataset.
+a :class:`kartothek.core.dataset.DatasetMetadata` object is returned. 
+This class holds information about the structure and schema of the dataset.
 
-The most interesting ones for this guide are ``tables`` and ``partitions``.
-Each dataset can have multiple tables, each table is a collection of files
-that all have the same schema. These files are called ``partitions`` in
-``kartothek``.
+For this guide, two attributes that are noteworthy are ``tables`` and ``partitions``:
 
-As we neither have explicitly defined the name of the table nor the name
+- Each dataset has one or more ``tables``, where each table represents a particular subset of
+  data, this data is stored as a collection of dataframes/files which have the same schema.
+- Data is written to storage in batches (for ``eager``, there is only a single batch),
```

Comment:
I wouldn't map the term `Batch` to a `Partition`, primarily because the batches as seen by the user are often different to the actual partitions. The ktk input is a batch as seen by the user but the partition is an implementation detail. In particular there are multiple possibilities for the mapping between the two:
- One input batch -> one partition
- One input batch -> multiple partitions
- Many input batches -> many partitions (e.g. when using dask shuffle)

While you are explaining a portion of the ambiguity here, the entirety is a bit more complex and beyond a getting started guide.
In this section you're introducing the term `Partition` which is a physical shard of the data, possibly for multiple tables. The origin of the data can have multiple source and I would simply state that this shard is not necessarily a one-to-one mapping to the input. The `partition_on` is a particular example where this is not the case. Instead of describing it's usage, I would forward to the section about the partitioning which describes an example where there is no one-to-one mapping.

Reply:
Fair point. Have attempted to simplify.